prow/spyglass/artifacts.go

```diff
@@ -50,29 +50,51 @@ func (s *Spyglass) ListArtifacts(ctx context.Context, src string) ([]string, err
 	}
 
 	artifactNames, err := s.StorageArtifactFetcher.artifacts(ctx, gcsKey)
-	logFound := false
-	for _, name := range artifactNames {
-		if name == "build-log.txt" {
-			logFound = true
-			break
-		}
+	if err != nil {
+		return artifactNames, fmt.Errorf("error retrieving artifact names from gcs storage: %v", err)
+	}
+
+	jobName, buildID, err := s.KeyToJob(src)
+	if err != nil {
+		return artifactNames, fmt.Errorf("error parsing src: %v", err)
 	}
-	if err != nil || !logFound {
-		artifactNames = append(artifactNames, "build-log.txt")
+
+	job, err := s.jobAgent.GetProwJob(jobName, buildID)
+	if err != nil {
+		return artifactNames, nil
+	}
+
+	jobContainers := job.Spec.PodSpec.Containers
+
+	if len(jobContainers) == 1 {
```

Comment:
I think it would be nice to combine the single container use-case with the for-loop. (At the least, you could move the if/else _inside_ the for-loop.)

Reply:
I am keeping a single container use-case just for backwards compatibility.  I will move it inside the for loop to be cleaner.